File: crypto/src/main/java/io/oversec/one/crypto/encoding/XCoderAndPadder.java
package io.oversec.one.crypto.encoding;

import io.oversec.one.crypto.encoding.pad.AbstractPadder;
import io.oversec.one.crypto.proto.Outer;


public class XCoderAndPadder {
    private final IXCoder mCoder;
    private final AbstractPadder mPadder;

    public XCoderAndPadder(IXCoder xCoder, AbstractPadder padder) {
        mCoder = xCoder;
        mPadder = padder;
    }

    public String getLabel() {
        return mCoder.getLabel(mPadder);
    }

    public String getExample() {
        return mCoder.getExample(mPadder);
    }

    public String getId() {
        return mCoder.getId() + (mPadder == null ? "" : ("__" + mPadder.getId()));
    }

    public AbstractPadder getPadder() {
        return mPadder;
    }

    public IXCoder getXcoder() {
        return mCoder;
    }

    public synchronized String encode(Outer.Msg msg, String srcText, boolean appendNewLines, String packagename) throws Exception {
        return mCoder.encode(msg, mPadder, srcText, appendNewLines, packagename);
    }

    public String getCoderId() {
        return mCoder == null ? null : mCoder.getId();
    }

    public String getPadderId() {
        return mPadder == null ? null : mPadder.getId();
    }
}

File: crypto/src/main/java/io/oversec/one/crypto/encoding/XCoderAndPadder.kt
package io.oversec.one.crypto.encoding

import io.oversec.one.crypto.encoding.pad.AbstractPadder
import io.oversec.one.crypto.proto.Outer

class XCoderAndPadder(val xcoder: IXCoder, val padder: AbstractPadder?) {

    val label: String
        get() = xcoder.getLabel(padder)

    val example: String
        get() = xcoder.getExample(padder)

    val id: String
        get() = xcoder.id + (padder?.let{ "__" + it.id} ?: "")

    val coderId: String?
        get() = xcoder.id

    val padderId: String?
        get() = padder?.id

    @Synchronized
    @Throws(Exception::class)
    fun encode(
        msg: Outer.Msg,
        srcText: String,
        appendNewLines: Boolean,
        packagename: String
    ): String {
        return xcoder.encode(msg, padder, srcText, appendNewLines, packagename)
    }
}
